### ISSUE TITLE
[Bug] Import of fields (permeability, porosity) using PAMELAMeshGenerator fails in parallel

### ISSUE THREAD

**Issue by francoishamon** [GEOSX/GEOSX#1082]
**Describe the bug**
In Tutorial 4, the permeability field is imported as follows:
```
    <PAMELAMeshGenerator
      name="mesh"
      file="../../../../../../../GEOSXDATA/DataSets/SPE10/EclipseBottomLayers/SPE10_LAYERS_83_84_85.GRDECL"
      fieldsToImport="{ PERM }"
      fieldNamesInGEOSX="{ permeability }"/>
```
On one MPI rank, everything works well, but on two ranks or more, some permeabilities have a zero value (and should not) when I inspect what is in the variable `property` of the following piece of code: 
https://github.com/GEOSX/GEOSX/blob/a56fdcf07049c059464f76d3c618af0fbe17d61c/src/coreComponents/meshUtilities/PAMELAMeshGenerator.cpp#L352
and then the code crashes. I was not able to track the problem further than that.

**To Reproduce**
1. Run Tutorial 4 from the `develop` branch on two MPI ranks,
2. Inspect the value of the property (here, permeability) that is imported.

**Expected behavior**
It should work.

**Platform:**
- LLNL/Quartz
- gcc 8.1.0
- develop, release build

**Additional context**
- Maybe this is related to #1041 but I am not sure since there is only one CellElementRegion here.
- I tried to find other tests where we import properties in parallel but did not find one

**Comment by francoishamon** [comment]
Hello @AntoineMazuyer,
Sorry to bring back this issue. I know that there is some work going on about a new mesher to replace PAMELA, so I was just wondering if you think it is still worth resolving this issue? I am asking because we now have multiple test cases for which we need to import a field in parallel.

Here is the problem: when we try to import a field using PAMELA (say, porosity or permeability), GEOSX works fine on one rank but fails in parallel because some values of permeability and/or porosity are not set by PAMELA. The issue appears for instance in Tutorials 4 and 5. For Tutorial 4 with 2 ranks, I get:
```
***** ERROR                                                                                                                                   
***** LOCATION: /g/g90/hamon1/geosx/GEOSX/src/coreComponents/meshUtilities/PerforationData.cpp:190                                            
***** Controlling expression (should be false): rEq < wellElemRadius[wellElemIndex]                                                           
***** Rank 0: The equivalent radius r_eq = 0 is smaller than the well radius (r = 0.1) in wellElementSubRegion  
```
The reason for this failure is that the well checks that `r_eq` (computed using the permeability) is strictly larger than 0. If I output the permeability, I see the following erroneous values causing `r_eq` to be zero:
```
perm[0][0][19799] = { 6.00949e-310, 5.51082e-311, 5.76971e-311 }                                                                              
perm[0][0][13197] = { 2.00041e-310, 2.00083e-310, 1.97494e-310 } 
```
I remember that you made a temporary fix for this in a separate PAMELA branch (`bugfix/mazuyer/import-ppt-mpi`). Is it worth merging this in the main branch?

**Comment by AntoineMazuyer** [comment]
Hello François,

I thought the branch you mentioned was merged.

I propose to open a PR and do tests to see if it's working. AFAIR we found a solution.

**Comment by francoishamon** [comment]
Hello @AntoineMazuyer,
I looked into the PAMELA code to try to fix the issue that prevents us from running Tutorials 4 and 5 in parallel. I ended up quite confused, so here is what I understood and the questions I have. The message is a little bit long so let's chat about this if it is easier.

It seems clear that the problem comes from here: https://github.com/GEOSX/PAMELA/blob/2720435d22f4ba2d70067764aca1e411c1274715/PAMELA/source/MeshDataWriters/MeshParts.cpp#L151
when we try to load a given mesh property (say, porosity). Let's consider a scalar property. If I understand correctly, `values` is a vector of size equal to the number of cells on one MPI rank, and `values_in_part` is a vector of size equal to the number of cells in the element region on one MPI rank.

1) In the main branch of PAMELA, we currently do this:
```
values_in_part[localIndex2*dimInt+i] = values[globalIndex*dimInt+i];
```
My understanding is that `localIndex2` is the local index of the cell in a given element region on one MPI rank, and but `globalIndex` is the global index of the cell in the entire domain (for all the element regions, for all the MPI ranks). This is not working in parallel because `values` is local to an MPI rank, but `globalIndex` spans the entire domain, so we go outside the bounds of the array and fill `values_in_part` with garbage.

2) In the branch `bugfix/mazuyer/import-ppt-mpi`, we do this:
```
values_in_part[localIndex2*dimInt+i] = values[localIndex2*dimInt+i + dimInt*offsets[curPart->Index]];
```
This fixes the problem for the tutorials (which have only one element region) but breaks the unit test `testPAMELAImport` because it does not work for the case of multiple element regions for two reasons. 
- First, `offsets` is a vector of size equal to the number of element regions plus one, but `curPart->Index` is the index of the region in the GMSH file, which produces a segfault. For instance, for the unit test `testPAMELAImport`, `curPart->Index = 103` for the `Reservoir` region, but `offsets` has a size equal to 5. I fixed this issue by adding an map from `curPart->Index` to an integer between 0 and the number of element regions.
- Second, using `offsets` like this assumes that the data in `values` is grouped by element regions, but I don't think this is the case. At least, I could not make it work.

So I am left with one question if you think there is hope to make this work: in PAMELA, is there a local index `localIndexOnMPIRank` that would allow us to do:
```
values_in_part[localIndexInRegionOnMPIRank*dimInt+i] = values[localIndexOnMPIRank*dimInt+i];
```
My understanding is that `localIndexInRegionOnMPIRank` already exists, but I could not find `localIndexOnMPIRank`. Sorry for bugging you with this and thanks for your help.
